### PR TITLE
fix(manager): removes outdated syncTarget field

### PR DESCRIPTION
**Pull request by danwt**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close https://github.com/dymensionxyz/dymint/issues/850

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Comment by omritoptix** [comment]
@danwt conflicts

File: block/pruning.go [hunk 1]
```diff
@@ -5,12 +5,6 @@ import (
 )
 
 func (m *Manager) pruneBlocks(retainHeight uint64) (uint64, error) {
-	syncTarget := m.SyncTarget.Load()
-
-	if retainHeight > syncTarget {
```
Comment (mtsitrin) [inline block/pruning.go:10]: @danwt why u removed this check?

Comment (danwt) [reply block/pruning.go:10]: It's not needed as far as I can tell, it's just an extra branch for no reason
I added a test to confirm

Comment (mtsitrin) [reply block/pruning.go:10]: needed for sequencer mode

File: block/manager.go [hunk 2]
```diff
@@ -170,8 +170,22 @@ func (m *Manager) Start(ctx context.Context) error {
...
+	slProposerKey := m.SLClient.GetProposer().PublicKey.Bytes()
+	localProposerKey, err := m.ProposerKey.GetPublic().Raw()
+	if err != nil {
+		return false, fmt.Errorf("get local node public key: %w", err)
+	}
+	return bytes.Equal(slProposerKey, localProposerKey), nil
+}
+
+func (m *Manager) IsSequencer() bool {
+	ret, _ := m.IsSequencerVerify()
```
Comment (mtsitrin) [inline block/manager.go:183]: bit redundent call no? why not just keep aggregator field on manager struct

Comment (danwt) [reply block/manager.go:183]: just prefer to minimize state
it's not an expensive call

File: block/block.go [hunk 3]
```diff
@@ -93,7 +93,9 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 	}
 
 	// Prune old heights, if requested by ABCI app.
-	if retainHeight > 0 {
+	wantToPrune := 0 < retainHeight
+	canPrune := !m.IsSequencer() || (uint64(retainHeight) <= m.NextHeightToSubmit()) // do not delete anything that we might submit in future
```
[the remark this answers is not in the document]
Comment (danwt) [reply block/block.go]: ok moved 👍

File: block/block.go [hunk 4]
```diff
@@ -93,11 +93,9 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 	}
 
 	// Prune old heights, if requested by ABCI app.
-	if retainHeight > 0 {
-		_, err := m.pruneBlocks(uint64(retainHeight))
-		if err != nil {
-			m.logger.Error("prune blocks", "retain_height", retainHeight, "err", err)
-		}
+	err = m.pruneBlocks(uint64(retainHeight))
```
Comment (mtsitrin) [inline block/block.go:96]: pls leave `if retainHeight > 0 {` (aka `wantToPrune`)
as it mostly disabled anyway, and it's better readability with the comment above

**Review by mtsitrin — CHANGES_REQUESTED** [review]

File: block/block.go (hunk 4 above)
Comment (danwt) [reply block/block.go]: done 👍

**Review by srene — APPROVED** [review]

**Review by omritoptix — COMMENTED** [review]
@danwt you may wanna change the filename of `synctarget`

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**